lib/galaxy/model/util.py

```diff
@@ -3,31 +3,94 @@
 """
 
 
+def count_toward_deleted_disk_usage(dataset_id, history_deleted, hda_deleted, dataset_deleted, dataset_disk_usage, **ids_dict):
```

Comment:
Do we need the kwargs dict passing here, or should it just be ids_dict (here and in the invocations) without any reference manipulation?

Reply:
The function needs to be called many times. I thought passing a reference might be more efficient than passing the whole dict. I could be wrong.

Reply:
Just a hunch, but I'd think passing the dict without the **kwargsing is faster.

Reply:
It's very clear to me yet, but I'll follow you.

Reply:
According to [https://stackoverflow.com/questions/13805461/makeing-a-list-of-dict-pass-by-value](https://stackoverflow.com/questions/13805461/makeing-a-list-of-dict-pass-by-value), we might keep **kwargsing here?

Reply:
@qiagu I'm not following your example, sorry.  Let's chat on gitter today to sort this out.